lib/logflare/context_cache.ex

```diff
@@ -32,43 +41,51 @@ defmodule Logflare.ContextCache do
         value
 
       {:ok, {:cached, value}} ->
-        index_keys(context, cache_key, value)
+        # already cached don't re-index
         value
     end
   end
 
+  @doc """
+  This function is called from the CacheBuster process when a new record comes in from the Postgres
+  write-ahead log. The WAL contains records. From those records the CacheBuster picks out
+  primary keys.
+
+  The records ARE the keys in the reverse cache (the ContextCache).
+
+  We must:
+   - Find the key by the record primary key
+   - Delete the reverse cache entry
+   - Delete the cache entry for that context cache e.g. `Logflare.Users.Cache`
+  """
+
   def bust_keys(values) do
-    {:ok, keys} = Cachex.keys(@cache)
-
-    total =
-      Enum.count(keys, fn {token, cache_key} = key ->
-        with true <- token in values,
-             {context, _} = token,
-             context_cache = cache_name(context),
-             {:ok, true} <- Cachex.del(context_cache, cache_key) do
-          Cachex.del(@cache, key)
-        end
-
-        true
-      end)
-
-    :telemetry.execute(
-      [:logflare, :context_cache, :busted],
-      %{count: total},
-      %{}
-    )
+    for {context, primary_key} <- values do
+      match_spec =
+        {:entry, {{context, primary_key}, :_}, :_, :_, :"$1"}
```

Comment:
Maybe using some library like [Matcha](https://github.com/christhekeele/matcha) or [ex2ms](https://github.com/ericmj/ex2ms) would make it a little bit cleaner?

Reply:
I was actually just thinking we should really be using fewer libs which is why I went direct 😭

Reply:
I agree with that sentiment. I was just wondering about others. I am pretty proficient with reading match specs, but others may not be.